src/lavinmq/shovel/shovel_store.cr

```diff
@@ -15,8 +15,33 @@ module LavinMQ
       end
     end
 
-    def create(name, config)
-      @shovels[name]?.try &.terminate
+    def upsert(name, config)
+      shovel = @shovels[name]?.try { |s| update_status_or_terminate(s, config) }
+      shovel ||= create(name, config)
+      shovel
+    end
+
+    private def update_status_or_terminate(shovel, config)
+      case {shovel.state, config["state"]?.try &.as_s}
+      when {"Running", "Paused"}
+        shovel.pause
+        return shovel
+      when {"Paused", "Running"}
+        spawn(shovel.resume, name: "Shovel name=#{shovel.name} vhost=#{@vhost.name}")
```

Comment:
1. Check wether this strategy is OK
2. If so, check wether we need this `as_s` on [shovel.cr](https://github.com/cloudamqp/lavinmq/blob/main/src/lavinmq/shovel/shovel.cr#L379)

Reply:
Honestly feels like Enum lose it's sense if they are treated as simple strings?

Reply:
Yeah, agree. I guess that state getter initially was only used as a display value in the mgmt ui? It should probably be changed.